src/main/java/com/excella/foodinc/controllers/FoodController.java

```diff
@@ -0,0 +1,38 @@
+package com.excella.foodinc;
+
+import java.util.HashMap;
+import java.util.Map;
+import reactor.core.publisher.Mono;
+
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class FoodController {
+
+    private static Map<String, Food> foodRepo = new HashMap<>();
+
+    static {
+        Food apple = new Food();
+        apple.setName("apple");
+        apple.setCalories(95);
+        apple.setFat(0);
+        apple.setProtein(0);
+        apple.setCarbs(5);
+        foodRepo.put(apple.getName(), apple);
+
+        Food pizza = new Food();
+        pizza.setName("pizza");
+        pizza.setCalories(200);
+        pizza.setFat(20);
+        pizza.setProtein(5);
+        pizza.setCarbs(30);
+        foodRepo.put(pizza.getName(), pizza);
+    }
+
+    @RequestMapping(value="/nutrition")
+    public Mono<Food> getFood(@RequestParam(value="food", defaultValue="apple") String name) {
+        return Mono.just(foodRepo.get(name));

```

Comment:
This is great, though you are directly talking to the data layer from the controller which should be modified so that the controller simply calls a service which then talks to the data layer